src/app/my-messages/todo-notification/todo-notification.component.spec.ts

```diff
@@ -0,0 +1,124 @@
+import { async, ComponentFixture, fakeAsync, TestBed } from '@angular/core/testing';
+
+import { TodoNotificationComponent } from './todo-notification.component';
+import { TodoNotification } from './todo-notification';
+import { NotificationType } from '../notification-type.enum';
+import { CommunityType } from '../../communities/community-type.enum';
+import { AppMaterialModule } from '../../app-material.module';
+import { BrowserAnimationsModule } from '@angular/platform-browser/animations';
+import { MatMomentDateModule } from '@angular/material-moment-adapter';
+import { ReactiveFormsModule } from '@angular/forms';
+import { RouterTestingModule } from '@angular/router/testing';
+import { CommunityRegistrationService } from '../../shared/community-registration.service';
+import { CommunityUserRegistrationResponse } from '../../shared/community-user-registration-response';
+import { of } from 'rxjs';
+import { CommunityUserRegistration } from '../../shared/community-user-registration';
+import { MatDialog } from '@angular/material';
+import { DeleteConfirmationDialogComponent } from '../../shared/delete-confirmation-dialog/delete-confirmation-dialog.component';
+import { BrowserDynamicTestingModule } from '@angular/platform-browser-dynamic/testing';
+
+const registrationResponse: CommunityUserRegistrationResponse = {
+  id: '567890',
+  user: {
+    id: '251c2a3b-b737-4622-8060-196d5e297ebc',
+    userName: 'testbed',
+    firstName: 'Tabia',
+    lastName: 'Testbed',
+    email: 'tabia.testbed@skoop.io'
+  },
+  approvedByUser: true,
+  approvedByCommunity: true
+};
+
+const currentUserId = 'e6b808eb-b6bd-447d-8dce-3e0d66b17759';
+const notification: TodoNotification = {
+  type: NotificationType.INVITATION_TO_JOIN_COMMUNITY,
+  id: currentUserId,
+  creationDatetime: new Date(),
+  communityName: 'Some closed community',
+  registration: {
+    id: 'b9f7a830-6437-4585-980c-b6820b6f03fb',
+    user: {
+      id: 'e6b808eb-b6bd-447d-8dce-3e0d66b17759',
+      userName: 'tester',
+      firstName: 'Toni',
+      lastName: 'Tester',
+      email: 'toni.tester@skoop.io'
+    },
+    approvedByUser: null,
+    approvedByCommunity: true,
+    community: {
+      id: '22c1ad17-4044-45a7-940c-22f1beeb7992',
+      title: 'Some closed community',
+      type: CommunityType.CLOSED,
+      description: 'Some closed community description',
+      links: []
+    }
+  }
+};
+
+describe('TodoNotificationComponent', () => {
+  let component: TodoNotificationComponent;
```

Comment:
If `TodoNotificationComponent` is turned into the abstract class `TodoNotificationComponentTrait` it will be impossible to instantiate it anymore. But one of the classes using the `TodoNotificationComponent` can still be instantiated and used for tests.